src/components/blocks/software/Code/MultilineCodeContent.tsx

```diff
@@ -44,19 +49,20 @@ export const MultilineCodeContent = ({
     contentWithObfuscatedKey,
   );
 
-  useLayoutEffect(() => {
-    Prism.hooks.add('before-highlight', function (env) {
-      env.code = env.element.textContent || '';
-    });
-
-    if (preRef.current) {
-      Prism.highlightElement(preRef.current);
-    }
-  }, [displayLanguage]);
+  const highlightedContent = useMemo(() => highlightSnippet(language, renderedContent), [language, renderedContent]);
```

Comment:
Was it not possible to highlight before the rendering stage?

Reply:
Sort of, would be a performance impact as we choose between different strings based on data provided

Reply:
Could you elaborate what kind of performance impact we're talking about here? It's important to remember that impact is now moved to the browser, i.e. the user. We don't track rendering perf (yet) but it would be interesting to see if this is noticeable on snippet heavy pages (maybe with some CPU throttling) and compare. (Ideally we'd have data for this already, but we don't).

(I wouldn't block this PR because of this, but I'd like for us to be confident that even on a device that is less powerful than an M1, the pages render fast-enough :) )

Reply:
I think the performance impact would be minimal, just seems wasteful to highlight very similar content multiple times and only display one version to me.

This is an abstract performance impact as in we're choosing which option 1 2 3 to render; if we run highlightSnippet(1) highlightSnippet(2) highlightSnippet(3), it can surely only be slower than choosing the option first and then highlighting it?

Reply:
Oh, I see what you mean;

it's not possible to render this without information from the user, so that's not possible.

Reply:
Yes, whether they're signed in or not and have API keys

Reply:
right, but that only means it needs to be interpolated later (realize this is not what's done now and might need rework. again, not advocating for anything like that now)

Reply:
Ticket created for this work